crates/telio-firewall/src/firewall.rs

```diff
@@ -567,13 +561,19 @@ impl StatefullFirewall {
                 telio_log_trace!("Connection {:?} closing", key);
                 if let Some(connection) = tcp_cache.get_mut(&key) {
                     connection.rx_alive = false;
-                    if !connection.tx_alive {
-                        telio_log_trace!("Removing TCP conntrack entry {:?}", key);
-                        tcp_cache.remove(&key);
-                    }
                 }
+            } else if !connection_info.tx_alive
+                && !connection_info.rx_alive
+                && !connection_info.conn_remote_initiated
+            {
+                if flags & TcpFlags::ACK == TcpFlags::ACK {
+                    telio_log_trace!("Removing TCP conntrack entry {:?}", key);
+                    tcp_cache.remove(&key);
+                    return true;
+                }
+                return false;
             }
-
+            tcp_cache.get(&key);
```

Comment:
Please add a comment explaining that the `get` is called here only to restart the cache entry timeout.

More importantly I think that this timeout update should only happen when `!(!connection_info.tx_alive && !connection_info.rx_alive && !connection_info.conn_remote_initiated)`

Reply:
>>> More importantly I think that this timeout update should only happen when !(!connection_info.tx_alive && !connection_info.rx_alive && !connection_info.conn_remote_initiated)

I think it already does that :smile: , but yeah, comment is missing, will add it

Reply:
You are right... I thought for some reason we are outside of the outer `if let Some...` :facepalm:

Reply:
updated